acvm-repo/acvm/src/pwg/memory_op.rs

```diff
@@ -74,11 +83,29 @@ impl<F: AcirField> MemoryOpSolver<F> {
         Ok(())
     }
 
+    /// Update the 'block_values' by processing the provided Memory opcode
+    /// The opcode 'op' contains the index and value of the operation and the type
+    /// of the operation.
+    /// They are all stored as 'Expressions'
```

Comment:
```suggestion
    /// They are all stored as an [Expression]
```
you could also link to the type itself in doc comments using something like above